kolibri/plugins/coach/assets/src/state/actions/reports.js

```diff
@@ -26,22 +26,99 @@ const ChannelResource = coreApp.resources.ChannelResource;
 const FacilityUserResource = coreApp.resources.FacilityUserResource;
 
 
+// helper function for showRecent, provides list of channels with recent activity
 function _showRecentChannels(store, classId) {
-  store.dispatch('CORE_SET_PAGE_LOADING', true);
-  store.dispatch('SET_PAGE_NAME', Constants.PageNames.RECENT);
-  const channelPromise = ChannelResource.getCollection();
-  channelPromise.fetch().then(
-    channels => {
+  function __getChannelLastActive(channel) {
```

Comment:
Lots of internal helper functions. Made sense to me, but let me know if this pattern is confusing